src/main/flight/gps_rescue.c

```diff
@@ -678,22 +664,13 @@ void descend(void)
     // set the altitude step, considering the interval between altitude readings and the descent rate
     float altitudeStepCm = taskIntervalSeconds * gpsRescueConfig()->descendRate;
 
-    // descend more slowly if the return home altitude was less than 20m
-    const float descentRateAttenuator = constrainf (rescueState.intent.returnAltitudeCm / 2000.0f, 0.25f, 1.0f);
-    altitudeStepCm *= descentRateAttenuator;
-    // slowest descent rate will be 1/4 of normal when we start descending at or below 5m above take-off point.
-    // otherwise a rescue initiated very low and close may not get all the way home
-
-    // descend faster while the quad is at higher altitudes
+    // descend faster while the quad is at higher altitudes, slower below 10m
     const float descentRateMultiplier = constrainf(rescueState.intent.targetAltitudeCm / 5000.0f, 0.0f, 1.0f);
```

Comment:
(Maybe skip that 2-step -process):
```suggestion
    const float descentRateMultiplier =  0.6f + (2.0f * constrainf(rescueState.intent.targetAltitudeCm / 5000.0f, 0.0f, 1.0f));
```

Or, maybe easier to understand:
```suggestion
    const float descentRateMultiplier = scaleRangef(constrainf(rescueState.intent.targetAltitudeCm, 0, 5000), 0,  5000, .6f, 2.6f);
```

Reply:
Unless I made some mistake, the result is exactly the same as your version. The idea is to make the code easier to understand.

`descentRateMultiplier`  was 0..1 value over  0..50m range. It is then mapped to .6 to 2.6 range on another line, when actually applying the value (so actual multiplier is .6 to 2.6. But why is variable named multiplier when it is not multiplying anything). 
With constrain+scaleRange, it is obvious that value is linear interpolation. 

There is no `two-stage descent with a faster phase and a slower phase` in either code ...

Reply:
With the minimum at 0.6, there is a two stage effect, because the descent rate is higher (up to 2.6 times max) at higher altitudes, but as it comes down, the descent rate is won't go below 0.6 times the max.  This happens at about 11.5m.  Hence the descent rate is faster above 11.5m and then steady below 11.5m.  That's what I mean by a two-stage process.

Your second code method is neater, thanks, I'll make it 0.6 and 3.0 so that the crossover is at 10m.

Reply:
I ran three methods through godbolt:
`0.6f + 2.4f * (fmax(0.0f, fmin(4000.0f, altitude - 1000.0f))) * 0.01f;`
`0.6f + 2.4f * constrainf(altitude - 1000.0f, 0.0f, 4000.0f) * 0.01f;`
`scaleRangef(constrainf(rescueState.intent.targetAltitudeCm, 1000, 5000), 1000, 5000, 0.6f, 3.0f);`

Either of the last two produce similar assembly, but your scaleRangef method was the most efficient, so I'll use that.  Thank you for your patience.

Reply:
Main advantage of `scaleRangef` is that it is easy to read.

Reply:
Yes definitely. Very nice